pkg/controller/endpointslice/utils.go

```diff
@@ -394,3 +394,16 @@ func hintsEnabled(annotations map[string]string) bool {
 	}
 	return val == "auto"
 }
+
+// managedByChanged returns true if one of the provided EndpointSlices is
+// managed by the EndpointSlice controller while the other is not.
+func managedByChanged(endpointSlice1, endpointSlice2 *discovery.EndpointSlice) bool {
+	return managedByController(endpointSlice1) != managedByController(endpointSlice2)
+}
+
+// managedByController returns true if the controller of the provided
+// EndpointSlices is the EndpointSlice controller.
+func managedByController(endpointSlice *discovery.EndpointSlice) bool {
```

Comment:
This seems to be reused in at least 2 more places, is not possible to export and consolidate them?

Reply:
I think this is actually the most straightforward given the reliance on the local `controllerName` var. Although we could extract this in the future, I'm fine with keeping it local for now.